chatminer/chatparsers.py

```diff
@@ -236,6 +236,47 @@ def _parse_message(self, mess):
         }
         return parsed_message
 
+class InstagramChatsParser(Parser):
+    def _read_file_into_list(self):
+        self._logger.info("Starting reading raw messages into memory...")
+        self.messages = []
+        with self._file.open(encoding="utf-8") as f:
+            messages_raw = reversed((json.load(f)["messages"]))
+
+        for line in messages_raw:
+            self.messages.append(line)
+        self._logger.info(
+            "Finished reading %i raw messages into memory.", len(self.messages)
+        )
+
+    def _parse_message(self, mess):
+        if "share" in mess:
+            if "link" in mess["share"]:
+                body = mess["share"]["link"]
+            elif "share_text" in mess["share"]:
+                body = mess["share"]["share_text"]
+            else:
+                body = "{} sent {}'s story.".format(mess["sender_name"], mess["share"]["original_content_owner"])
+        elif "photos" in mess:
+            body = mess["photos"][0]["uri"]
+        elif "videos" in mess:
+            if len(mess["videos"]):
+                body = mess["videos"][0]["uri"]
+            else:
+                body = "Photo sent by {}".format(mess["sender_name"])
```

Comment:
When is this executed? If there's a `videos` field in the message but it's empty? Don't we catch `photos` in the `elif` before?

Reply:
Oops! That was a typo and was supposed to be "Video sent by...". As for when it's executed, I found a strange bug where sometimes there would be a video object with nothing in it. It only happened once, but it caused the program to stop. I will remove this and add a warning instead, because after testing this it seems like adding an entry isn't the best solution.